tari-project/universe
# feat: download and run vc installer ( fix part 2/2 )

# Second part of fix: #725 

### [ Features WINDOWS ONLY ]
- Improved reading of installed applications on system
- Extended list of required external dependencies on system
- Improved detection if user have all required dependecies
- Implemented dialog for displaying external dependencies with information about package and its manufacturer
- Provided button which downloads and starts installer for the package

------------
### [ Notes ]
From what I've been testing gpu miner:
- Will work fine with `2022` and `2019` packages
- Throw `non-zero exit code Some(-1073741515` for versions `2017` and `2015`
Going with lower versions like `2012` will cause to app not starting at all and throwing error about missing msvcp140.dll
-  When using Microsoft Visual C++ Redistributable with version lower then from year 2019 I cannot start app in dev mode
------------
### [ Testing ]

- Remove all Microsoft Visual C++ Redistributable packages from system
- Install for example Microsoft Visual C++ Redistributable 2015 https://www.microsoft.com/pl-pl/download/details.aspx?id=48145
- After opening application you should see dialog with statuses of external dependencies
![image](https://github.com/user-attachments/assets/f5848c61-f0b6-4049-8282-70ba44ef4169)
- After clicking button `download and install` and successfully installing package its status should turn to `Installed`
- Click continue to resume setup
------------
### [ Issues ]
- On my machine I found that without any Microsoft Visual C++ Redistributable package I cannot even start application, getting error of missing msvcp140.dll
- Sometimes when clicking `download and install` it can throw error about problem with downloading package, it does not crash the app and user can click again which most of time download and run install successfully
------------
### [ Wideo with happy path ]
[happy_path.zip](https://github.com/user-attachments/files/17295136/happy_path.zip)
